odoo/models.py

```diff
@@ -349,6 +349,16 @@ def expand_ids(id0, ids):
     'year': dateutil.relativedelta.relativedelta(years=1)
 }
 
+READ_GROUP_NUMBER_GRANULARITY = {
+    'quarter_number': 'quarter',
+    'month_number': 'month',
+    'week_number': 'week',
```

Comment:
Why don't you also add `year_number`, `day_number` and `hour_number`, for the sake of uniformity with the absolute time granularities?
When using `_read_group()` programmatically, grouping by `year` or `year_number` are equivalent, but the returned type is different: `year` will return a date/datetime, while `year_number` will return a number, which might be much more convenient.
And it really does not cost anything to add them, so why not?